src/compiler/types.ts

```diff
@@ -2089,6 +2108,7 @@ namespace ts {
         ESSymbol                = 0x01000000,  // Type of symbol primitive introduced in ES6
         ThisType                = 0x02000000,  // This type
         ObjectLiteralPatternWithComputedProperties = 0x04000000,  // Object literal type implied by binding pattern has computed properties
+        PredicateType           = 0x08000000,

```

Comment:
Fixed the issue - it was simply that `this` type guards weren't re-instantiated by `instantiateType`. :+1: